src/Console/Command/GenerateExtensionCommand.php

```diff
@@ -10,116 +10,367 @@
+            passthru('cd '.escapeshellarg($dir).'; cd js/forum; npm install; gulp default; cd ../admin; npm install; gulp default');
+        }
 
-        passthru("cd $dir; composer install; cd js/forum; npm install; gulp; cd ../admin; npm install; gulp");
+        $this->info('Updating Composer');
 
-        $this->info('Extension "'.$name.'" generated!');
+        if (! isset($composer['repositories'])) {
+            $composer['repositories'] = [
+                [
+                    'type' => 'path',
+                    'url' => 'workbench\/*\/',
+                ],
+            ];
+        }
+
+        $composer['require'][$packageName] = 'dev-master';
+
+        file_put_contents($composerPath, json_encode($composer, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+        passthru('cd '.escapeshellarg(public_path()).'; composer update');
+
+        $this->getApplication()->find('cache:clear')->run($this->input, $this->output);
+
+        $this->info('Extension "'.$package.'" generated!');
     }
 
+    /**
+     * Returns option value otherwise ask user to input.
+     *
+     * @param string $question
+     * @param callable $checkValid
+     * @param string|null $default
+     * @return string
+     */
+    protected function getOptionOrAsk($option, $question, callable $checkValid, $default = null)
+    {
+        $input = $this->input->getOption($option);
+
+        if ($input) {
+            if (! $checkValid($input)) {
+                throw new Exception($option.' option is not valid');
+            }
+
+            return $input;
+        }
+
+        do {
+            $input = $this->ask($question, $default);
+        } while (! $checkValid($input));
+
+        return $input;
+    }
+
+    /**
+     * Ask user to input string.
+     *
+     * @param string $question
+     * @param string|null $default
+     * @return string
+     */
     protected function ask($question, $default = null)
     {
-        $question = new Question("<question>$question</question> ", $default);
+        $question = new Question(
+            '<question>'.$question.($default ? ' <comment>['.$default.']</comment>' : '').':</question> ',
+            $default
+        );
 
         return $this->getHelperSet()->get('question')->ask($this->input, $this->output, $question);
     }
 
+    /**
+     * Copy extension skeleton to destination.
+     *
+     * @param string $destination
+     * @param array $replacements
+     * @return void
+     */
     protected function copyStub($destination, $replacements = [])
     {
         $this->recursiveCopy(__DIR__.'/../../../stubs/extension', $destination, $replacements);
     }
 
+    /**
+     * Copy source to destination.
+     *
+     * @param string $src
+     * @param string $dst
+     * @param array $replacements
+     * @return void
+     */
     protected function recursiveCopy($src, $dst, $replacements = [])
     {
         $dir = opendir($src);
-        @mkdir($dst);
+        @mkdir($dst, 0775, true);
```

Comment:
See my comment on #915 ; using `@` suppresses any error messages when creating a directory. The code should however check for its return code (true or false) whether it was succesful.